src/nnue/nnue_common.h

```diff
@@ -162,77 +166,118 @@ namespace Stockfish::Eval::NNUE {
               write_little_endian<IntType>(stream, values[i]);
   }
 
+
+  // read_leb_128(s, out, N) : read N signed integers from the stream s, putting them in
+  // the array out. The stream is assumed to be compressed using the signed LEB128 format.
+  // See https://en.wikipedia.org/wiki/LEB128 for a description of the compression scheme.
   template <typename IntType>
   inline void read_leb_128(std::istream& stream, IntType* out, std::size_t count) {
-      static_assert(std::is_signed_v<IntType>, "Not implemented for unsigned types");
+
+      // check that presence of our LEB128 magic string
```

Comment:
If we're doing any work around here, then I'd suggest we move this check to a separate function that calls read_leb_128. The rationale is that our magic string is a property of tensor serialization, not of the LEB128 compressed data. I envision a `read_tensor` function that checks the magic string and calls `read_leb_128`. Possibly not in scope of this PR.

Reply:
yes, probably better to do it in another PR, if any